File: frontend/src/app/service/comment/comment.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {CommentModel} from "../../model/commentmodel";

@Injectable({
  providedIn: 'root'
})
export class CommentService {

  constructor(private http: HttpClient) {
  }

  findCommentsByTaskId(id: string): Observable<CommentModel[]> {
    return this.http.get<CommentModel[]>('/api/comments/taskId/' + id);
  }

  saveComment(comment: CommentModel): Observable<CommentModel> {
    return this.http.post<CommentModel>("/api/comments", comment);
  }
}

File: fapi/src/main/java/com/netcracker/edu/project/fapi/service/CommentService.java
package com.netcracker.edu.project.fapi.service;

import com.netcracker.edu.project.fapi.model.CommentModel;

import java.util.List;

public interface CommentService {
    List<CommentModel> findCommentsByTaskId(Integer id);
    CommentModel saveComment(CommentModel comment);
}
